### PR TITLE
feat: add a wrapper for encryption and decryption

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Added a wrapper for Crypto Operation (Encrypt, Decrypt). So any changes in Encryption framework can be done in one place. This also adds logging for table name for which the Encryption and Decryption is happening

## Motivation and Context

This adds two things

- A wrapper for encryption and decryption operation
- A table name field for encryption and decryption logs

## How did you test it?

- All sanity tests for MerchantAccount Create, Payments, Refunds
- Check the table name field in the encryption logs

Screenshot 2024-08-07 at 2 49 36 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x]  I reviewed the submitted code

### CODE REVIEW

File: crates/router/src/db/merchant_connector_account.rs [hunk 1]
```diff
@@ -1100,13 +1104,15 @@ mod merchant_connector_account_cache_tests {
         let mca = domain::MerchantConnectorAccount {
             merchant_id: merchant_id.to_owned(),
             connector_name: "stripe".to_string(),
-            connector_account_details: domain::types::encrypt(
+            connector_account_details: domain::types::crypto_operation(
                 key_manager_state,
-                serde_json::Value::default().into(),
+                type_name!(domain::MerchantKeyStore),
```
Comment (ArjunKarthik) [inline crates/router/src/db/merchant_connector_account.rs:1109]: MerchantConnectorAccount

File: crates/router/src/core/payment_methods/cards.rs [hunk 2]
```diff
@@ -4241,26 +4262,29 @@ async fn get_bank_account_connector_details(
...
-                .change_context(errors::StorageError::DeserializationFailed)
-                .change_context(errors::ApiErrorResponse::InternalServerError)
-                .attach_printable("Failed to deserialize Payment Method Auth config")
-        },
-    )
-    .transpose()?;
+    let payment_method_data =
+        domain::types::crypto_operation::<serde_json::Value, masking::WithType>(
+            &state.into(),
+            type_name!(domain::MerchantKeyStore),
```
Comment (ArjunKarthik) [inline crates/router/src/core/payment_methods/cards.rs:4268]: PaymentMethod

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by vspecky — APPROVED** [review]
LGTM on behalf of @juspay/hyperswitch-routing

**Review by racnan — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by vspecky — APPROVED** [review]
LGTM on behalf of @juspay/hyperswitch-routing

**Review by jarnura — APPROVED** [review]

**Review by racnan — APPROVED** [review]

**Pull request merged**